testplan/common/utils/networking.py

```diff
@@ -8,10 +8,105 @@
     absolute_import,
 )
 import socket
+import random
+from six.moves import range
 
 
 def port_to_int(port):
     try:
         return int(port)
     except ValueError:
         return socket.getservbyname(port)
+
+
+def unoccupied_ports(port_low_high, nb_ports):
+    """
+    Finds free TCP ports in the given range by binding to random ports in the range,
+    until <nb_ports> free ports have been found.
+
+    This does not guarantee that by the time caller tries to bind the ports will
+    still be available.
+
+    .. warning::
+
+        You should *not* use this utility. Ever. That is because it is impossible to guarantee
```

Comment:
I don't think we should be porting this utility here if we can avoid it - as the warning says, it is not a reliable way of acquiring free ports.